pkg/tests/end_to_end_tests/trace_query_integration_test.go

```diff
@@ -0,0 +1,414 @@
+		require.NoError(t, err)
+		jTraces := convertToTraces(jaegerResp.Data.([]interface{}))
+
+		// Sort traces.
+		sort.SliceStable(pTraces, func(i, j int) bool {
+			return pTraces[i].TraceID < pTraces[j].TraceID
+		})
+		sort.SliceStable(jTraces, func(i, j int) bool {
+			return jTraces[i].TraceID < jTraces[j].TraceID
+		})
+
+		require.Equal(t, len(jTraces), len(pTraces), "num of traces returned in fetch traces API")
+
+		// Sort the trace contents and prepare them to compare.
+		for i := 0; i < len(jTraces); i++ {
+			a := &jTraces[i]
+			b := &pTraces[i]
+
+			require.Equalf(t, len(a.Spans), len(b.Spans), "num spans of %d", i)
+			require.Equalf(t, a.TraceID, b.TraceID, "trace id of %d", i)
+
+			sortTraceContents(a)
+			sortTraceContents(b)
+
+			purgeProcessInformation(a)
+			purgeProcessInformation(b)
+		}
+		require.Exactly(t, jTraces, pTraces)
+	})
+}
+
+func insertDataIntoJaeger(endpoint string, data pdata.Traces) error {
+	opts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithBlock()}
+	conn, err := grpc.Dial(endpoint, opts...)
+	if err != nil {
+		panic(err)
+	}
+
+	client := jaegerproto.NewCollectorServiceClient(conn)
+
+	batches, err := jaegertranslator.InternalTracesToJaegerProto(data)
+	if err != nil {
+		panic(err)
+	}
+
+	for _, batch := range batches {
+		_, err := client.PostSpans(context.Background(), &jaegerproto.PostSpansRequest{Batch: *batch}, grpc.WaitForReady(true))
+		if err != nil {
+			panic(err)
+		}
+	}
+	return nil
+}
+
+var skipTags = map[string]struct{}{
+	"internal.span.format": {},
+	"otel.status_code":     {},
+}
+
+func sortTraceContents(t *jaegerJSONModel.Trace) {
+	sort.Strings(t.Warnings)
+
+	sortSpanContents(t.Spans, skipTags)
+}
+
+func sortSpanContents(spans []jaegerJSONModel.Span, ignoreTags map[string]struct{}) {
+	sort.SliceStable(spans, func(i, j int) bool {
+		return spans[i].SpanID < spans[j].SpanID
+	})
+
+	for index := range spans {
+		// Sort references.
+		sort.SliceStable(spans[index].References, func(i, j int) bool {
+			return spans[index].References[i].SpanID < spans[index].References[j].SpanID
+		})
+
+		spans[index].Tags = sortFields(trimIgnoredTags(spans[index].Tags[:], ignoreTags))
+
+		// Sort warnings.
+		sort.Strings(spans[index].Warnings)
+
+		if spans[index].Process != nil {
+			// Sort processes tags.
+			spans[index].Process.Tags = sortFields(spans[index].Process.Tags[:])
+		}
+
+		// Sort tags in each log.
+		for logIndex := range spans[index].Logs {
+			// Sort tags belonging to a log.
+			spans[index].Logs[logIndex].Fields = sortFields(spans[index].Logs[logIndex].Fields[:])
+		}
+	}
+}
+
+func sortFields(f []jaegerJSONModel.KeyValue) []jaegerJSONModel.KeyValue {
+	sort.Slice(f, func(i, j int) bool {
+		v1 := fmt.Sprintf("%s%v", f[i].Key, f[i].Value)
+		v2 := fmt.Sprintf("%s%v", f[j].Key, f[j].Value)
+		return v1 < v2
+	})
+	return f
+}
+
+func purgeProcessInformation(a *jaegerJSONModel.Trace) {
+	a.Processes = nil
```

Comment:
can we just purge the ProcessID instead of the entire process object?

Reply:
I don't think so, as its a map where the process information can be diff b/t two responses.

Reply:
Can't we just sort the map values an renumber with a processID starting at 0?

Reply:
AFAIK the order of map was not the problem (if it was, reflect.DeepEqual will not complain), but its contents, are dissimilar. 

Jaeger returns
```
Processes:map[json.ProcessID]json.Process(nil)
```

Promscale returns
```
Processes:map[json.ProcessID]json.Process{"p1":json.Process{ServiceName:"service-name-0", Tags:[]json.KeyValue{json.KeyValue{Key:"resource-attr", Type:"string", Value:"resource-attr-val-0"}}}
```

Feels like the wrapper needs some more things. But nothing as a bug, since it's not Promscale code anyways. So, if we spot a better way of the wrapper, we can optimize this in future.